src/java/org/apache/cassandra/utils/JVMStabilityInspector.java

```diff
@@ -186,17 +205,39 @@ public static void userFunctionTimeout(Throwable t)
         {
             case die:
                 // policy to give 250ms grace time to
-                ScheduledExecutors.nonPeriodicTasks.schedule(() -> killer.killCurrentJVM(t), 250, TimeUnit.MILLISECONDS);
+                ScheduledExecutors.nonPeriodicTasks.schedule(() -> killer.killJVM(t), 250, TimeUnit.MILLISECONDS);
                 break;
             case die_immediate:
-                killer.killCurrentJVM(t);
+                killer.killJVM(t);
                 break;
             case ignore:
                 logger.error(t.getMessage());
                 break;
         }
     }
 
+    public static void registerShutdownHook(Thread hook, Runnable runOnHookRemoved)
+    {
+        Runtime.getRuntime().addShutdownHook(hook);
+        shutdownHooks.add(Pair.create(hook, runOnHookRemoved));
+    }
+
+    public static void removeShutdownHooks()
+    {
+        Throwable err = null;
+        for (Pair<Thread, Runnable> hook : shutdownHooks)
+        {
+            err = Throwables.perform(err,
+                                     () -> Runtime.getRuntime().removeShutdownHook(hook.left),
+                                     hook.right::run);
+        }
+
+        if (err != null)
+            logger.error("Got error when removing shutdown hook(s): {}", err.getMessage(), err);
```

Comment:
Is it correct to log the error if `killJVM` was called to be quiet? To me it sounds as violation of the [interface description](https://github.com/datastax/cassandra/pull/271/files#diff-eadfd87d7242f6ef077d475dd3d9a4e4240ddd843ac420d0f0d0bd9c3c25da8cR41):
> @param quiet - whether the error should be logged verbosely